src/Utils.ts

```diff
@@ -30,8 +30,34 @@ export const planckToUnit = (val: number, units: number): number => {
 };
 
 export const planckBnToUnit = (val: BN, units: number): number => {
-  const value = val.toNumber() / 10 ** units;
-  return value;
+  const Bn10 = new BN(10);
+  const BnUnits = new BN(units);
+  const div = val.div(Bn10.pow(BnUnits));
+  const mod = val.mod(Bn10.pow(BnUnits));
+  const whole = div.toString();
+  const decimal = mod.toString().padStart(units, '0');
+  const result = `${whole}.${decimal || '0'}`;
+  return Number(result);
+};
+
+export const unitToPlanckBn = (val: number, units: number): BN => {
+  // conver to number in case the number arguments are passed as numeric strings
+  val = Number(val);
+  units = Number(units);
+
+  const Bn10 = new BN(10);
+  const BnUnits = new BN(units);
+  console.log(val);
+  const [w, d] = val.toFixed(units).split('.');
+  const wholeBn = new BN(Number(w));
+  const decimalBn = new BN(Number(d));
+  const resultBn = wholeBn.mul(Bn10.pow(BnUnits)).add(decimalBn);
```

Comment:
Would it be possible to comment this process out?

Reply:
Just to clarify are you asking to remove this method or add comments/docs :) It is used for converting the funds from chain unit to BN chain decimal/planck. I can add proper comments/docs.

Reply:
Just to add comments :)

Reply:
I added some comments for chain decimal calculations. Feel free to add/edit:)